src/qtAliceVision/SequenceCache.cpp

```diff
@@ -260,55 +270,79 @@ void SequenceCache::onPrefetchingProgressed(int)
     Q_EMIT requestHandled();
 }
 
-void SequenceCache::onPrefetchingDone(int reqFrame)
+void SequenceCache::onPrefetchingDone(int sequenceId, int reqFrame)
 {
-    // Update internal state
-    _loading = false;
+    //Make sure the fetching concerns the actual sequence Id
+    bool exitOld = false;
+    _lockSequence.lock();
+    if (sequenceId != sequenceId)
```

Comment:
I think there might be a typo: shouldn't it be `sequenceId != _sequenceId` instead ?

Reply:
yes. wouhou :/